### PR TITLE
Detected nucleus model

### PR DESCRIPTION
**Pull request by annehaley**

### CODE REVIEW

**Comment by naglepuff** [comment]
I'm curious: if you were to run `generateTypes`, what would the `Dataset` interface look like? Would there be an optional `detected_nuclei?: DetectedNucleus` field?

**Comment by naglepuff** [comment]
I'm curious: if you were to run `generateTypes`, what would the `Dataset` interface look like? Would there be an optional `detected_nuclei?: DetectedNucleus` field?

**Comment by annehaley** [comment]
> I'm curious: if you were to run `generateTypes`, what would the `Dataset` interface look like? Would there be an optional `detected_nuclei?: DetectedNucleus` field?

Yes, that's how it should generate the type (though I'm not sure of the actual behavior). We can expect this field to exist on Datasets of the type `detected_nuclei`. Other types will have this field removed because it is not applicable.

File: atlascope/core/job_types/nucleus_detection.py
Comment (waxlamp) [inline atlascope/core/job_types/nucleus_detection.py]: This way of expressing it is a bit odd, and took me some reading of different parts of the code to understand what's going on. At the very least, I think you should factor this expression out to before the call to `create()` and construct the dict with an expressive name, then "dump" that object here with the `**` operator.

This is purely a readability thing, but that is an important consideration in complex code such as this.

**Review by waxlamp — APPROVED** [review]
If you can refactor the code as I suggested, then there's no need for a second review and you can merge as soon as that's done. Thanks!

File: atlascope/core/job_types/nucleus_detection.py
Comment (annehaley) [reply atlascope/core/job_types/nucleus_detection.py]: Ok, I changed this in [1394483](https://github.com/atlascope/atlascope/pull/169/commits/139448398be68ecada9512934a276c1ac7b11a16)

**Pull request merged**
